core/src/tile/tileManager.cpp

```diff
@@ -473,11 +460,12 @@ void TileManager::updateProxyTiles(TileSet& _tileSet, const TileID& _tileID, Til
         return;
     }
     // Try children
-    if (m_view->s_maxZoom > _tileID.z) {
-        for (int i = 0; i < 4; i++) {
-            updateProxyTile(_tileSet, _tile, _tileID.getChild(i), static_cast<ProxyID>(1 << i));
-        }
+    // FIXME maxZoom also depends on DataSource
```

Comment:
Yup, good catch, view maxZoom and dataSource maxZoom can be different. A legit FIXME :D.